score-server/src/main/java/bio/overture/score/server/config/SecurityConfig.java

```diff
@@ -163,4 +130,77 @@ private boolean useJwt(HttpServletRequest request) {
     }
     return true;
   }
+
+  @Bean
+  public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+    http.authorizeHttpRequests(
+            authorize ->
+                authorize
+                    .antMatchers("/isAlive")
+                    .permitAll()
+                    .antMatchers("/studies/**")
+                    .permitAll()
+                    .antMatchers("/upload/**")
+                    .permitAll()
+                    .antMatchers("/entities/**")
+                    .permitAll()
+                    .antMatchers("/export/**")
+                    .permitAll()
+                    .antMatchers("/schemas/**")
+                    .permitAll()
+                    .antMatchers(swaggerConfig.getAlternateSwaggerUrl())
+                    .permitAll()
+                    .antMatchers(
+                        "/swagger**",
+                        "/swagger-ui.html",
+                        "/swagger-ui**",
+                        "/swagger-resources/**",
+                        "/v2/api-docs/**",
+                        "/webjars/**")
+                    .permitAll()
+                    .anyRequest()
+                    .authenticated())
+        .oauth2ResourceServer(
+            oauth2 -> oauth2.authenticationManagerResolver(tokenAuthenticationManagerResolver()));
+
+    return http.build();
+  }
+
+  @Bean
+  public AuthenticationManagerResolver<HttpServletRequest> tokenAuthenticationManagerResolver() {
+    AuthenticationManager jwtManager =
+        new ProviderManager(new JwtAuthenticationProvider(jwtDecoder));
+
+    AuthenticationManager authzOpaqueManager =
+        new ProviderManager(new OpaqueTokenAuthenticationProvider(authzTokenIntrospector));
+
+    return (request) -> {
+      if (isAuthZRequest(request)) {
+        return authzOpaqueManager;
+      } else {
+        return jwtManager;
+      }
+    };
+  }
```

Comment:
Choose token manager based on auth provider, like in `file-manager`.

Reorder the if check so that we are not initializing a class we are not going too use or return.